README.md

```diff
@@ -95,6 +95,27 @@ Install [jq](https://stedolan.github.io/jq) so that the parameters can get parse
 make test
```

Comment:
General notices.
1. Why `WithContext` is not enough for you?
2. You know, that this works only if `WithContext` is used before?

Reply:
I updated all logs to include `WithContext` where a context is passed so that this will work in all possible places. I think the thing that is nice about this is that by setting the tags on init, it gives the client an easy method to ensure that all logs have the same tag, rather than needing to wrap each call. Does that seem reasonable?

Reply:
I'm not sure I understand. When you add `WithContext`, all fields defined in `LogKeys` variable is used. Maybe we don't need two mechanisms of this, maybe it would be better to define something like:

```
func registerContextVariableToLog(name string) {
    LogKeys = append(LogKeys, name)
}
```

WDYT?

Reply:
Ah I wasnt aware I could use this since LogKeys is currently configured to be an array of fixed length (set to 2). Can I alter it to be an array of any length?

Reply:
One other thing to note is that we typically use struct{} instead of string as the type to set context keys, since there are some benefits (compiler will check for duplicate keys, type safety etc) so maybe LogKeys could take in any?

Reply:
Ad 1 - yes, please do.
Ad 2 - yes, I think you can. Can you please add some tests to `TestLogEntryCtx`?

Reply:
Now that I look into it more, the usage of the contextKey as the key in the log will cause problems if we allow the LogKeys to be type any. I will update LogKeys to be a map of string to type contextKey where contextKey is any. I will add a test to confirm that it works as expected

Reply:
I don't think we should do that. This may be an incompatible change in public API. So we should either stick to array of strings or use something more generic.

Reply:
Ok, what do you think of not changing the type of contextKey, but having LogKeys map string -> any? I added a test to confirm that the behavior of `context2Fields` does not change if we try it with that

Reply:
Yes, that seems backward compatible :)

Reply:
Unfortunately, changing a type from list to map is still backward incompatible :) What I mean is that `LogKeys` is a publicly accessible variable, meaning that our clients may change it. So if we change the type of this field, their code won't compile anymore.

I'm wondering about root of what you need. Is this just logging specific context key/value or is this something more? If it is only logging key/value, why can't you just add additional value to `LogKeys` in your app?

Reply:
Ah I see. The use case for us is that since its the go recommendation to use structs as the contextKeys, rather than string, and since LogKeys takes in the key to search the context for, we are not able to use the map because of the type restriction. All of our keys are structs, so that was where the initial idea of having the separate data store came from

Reply:
Ahh, I can finally see it! So because of strong typing, we can't modify `LogKeys` type. In that case, the solution you proposed previously was proper. Apologies for not understanding it earlier!

So in my opinion the best one was with registering hook - log key and function retrieving from context. So it should be probably something like
```
func RegisterLogContextHook(contextKey, ctxExtractor func(ctx contex.Context) string)
```

Reply:
Ok thank you for all your help! Making the update now!